assets/js/base/context/checkout-context.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import {
+	createContext,
+	useContext,
+	useState,
+	useMemo,
+} from '@wordpress/element';
+
+const CheckoutContext = createContext( {} );
+
+export const useCheckoutContext = () => {
+	return useContext( CheckoutContext );
+};
+
+const CheckoutProvider = ( { children, initialActivePaymentMethod } ) => {
+	const [ successRedirectUrl, setSuccessRedirectUrl ] = useState( '' );
+	const [ failureRedirectUrl, setFailureRedirectUrl ] = useState( '' );
+	const [ checkoutComplete, setCheckoutComplete ] = useState( false );
+	const [ checkoutHasError, setCheckoutHasError ] = useState( false );
```

Comment:
hmm ya that sounds like a better approach - the only concern I might have is if there are multiple errors but that would likely typically happen with form field validation (in which case there might be multiple field validation fields but there would be only one error message).  Validation interface hasn't been worked out here yet but I'll take your feedback in consideration when that's worked on.  Likely will incorporate your suggestion.

Reply:
I'm going to leave this as is for now.  The thinking is that when there is an error, two things will happen:

- An error notice will be created and added to the notices via `addNotice` (and would get displayed)
- The checkout will be set to an error state via `setHasError`.  The error state will remain until conditions introducing that error are cleared.

There's still some ambiguity around what exactly will be needed and how this will actually function so I think it'd be good to leave as is until actually implementing some different error states etc and seeing how that actually works.